source/cellular_bg96_api.c

```diff
@@ -2223,6 +2321,39 @@ static CellularPktStatus_t socketSendDataPrefix( void * pCallbackContext,
     return pktStatus;
 }
 
+static CellularPktStatus_t fileUploadInFileSystem( void * pCallbackContext,

```

Comment:
I'm not sure I understand this function... From what I gather you're putting a "\n" at the second position of pLine. I reckon that is likely because of the comment in lines 2346-47 ... Consider improving that comment to provide a bit more context. Anyway, good comment, it's already providing a lot of context.

Reply:
After a "CONNECT sent by the modem, the data is uploaded in the next line, therefore the "\n" is added.